frost-ed25519/tests/integration_tests.rs

```diff
@@ -8,6 +8,15 @@ fn check_zero_key_fails() {
     frost_core::tests::ciphersuite_generic::check_zero_key_fails::<Ed25519Sha512>();
 }
 
+#[cfg(feature = "cheater-detection")]
+#[test]
+fn check_sign_with_dkg() {
+    let rng = thread_rng();
+
+    frost_core::tests::ciphersuite_generic::check_sign_with_dkg::<Ed25519Sha512, _>(rng);
+}
+
+#[cfg(not(feature = "cheater-detection"))]
```

Comment:
I don't get this, both options seems to be the same? Maybe leftover code?

Reply:
They all call check_sign() in the tests which is where the feature is toggled. https://github.com/ZcashFoundation/frost/blob/cheater_detection_355/frost-core/src/tests/ciphersuite_generic.rs#L251. So these tests are one with the feature enabled and one without. Is there a nicer way of writing this?

Reply:
All these functions call check_sign() which has the feature toggled

Reply:
Ah, I see what was your intention. This will not quite work that way, the `cfg()` attribute will just enable some piece of code depending on whether the feature was enabled or not, it won't actually enable or disable it.

Here both functions are the same, so there is no need to gate them under `cfg()`. Whether it will call the `check_sign()` with the feature enabled or not will only depend on whether the feature is enabled in the first place.

So you can remove the `cfg()` gate there and leave it like it was before.

In order to actually test the case where the feature is disabled, we'll need to call `cargo test --release --no-default-features` in CI (by adding a new entry in `main.yml`). You can do that in this PR or in a separate one, up to you.

(Ideally we should test all possible combinations of features. I think there is a cargo plugin that helps with that. But that's probably overkill in our case)

Reply:
Ah ok. Thank you. I'll update the code